packages/ra-core/src/controller/list/useSelectAll.tsx

```diff
@@ -0,0 +1,117 @@
+import { useCallback } from 'react';
+import { useQueryClient } from '@tanstack/react-query';
+
+import { useNotify } from '../../notification';
+import { useDataProvider, UseGetListOptions } from '../../dataProvider';
+import type { FilterPayload, RaRecord, SortPayload } from '../../types';
+import type { UseRecordSelectionResult } from './useRecordSelection';
+
+/**
+ * Function hook to select all items of a list (until we reached the limit)
+ *
+ * @param {string} resource Required. The resource name
+ * @param {RecordSelectionModifiers} selectionModifiers Required, The selection modifiers
+ * @param {SortPayload} sort Optional. The sort object passed to the dataProvider
+ * @param {FilterPayload} filter Optional. The filter object passed to the dataProvider
+ * @returns {Function} onSelectAll A function to select all items of a list
+ *
+ * @example
+ *
+ * const MyButton = () => {
+ *   const [_, selectionModifiers] = useRecordSelection({ resource: 'posts' });
+ *   const onSelectAll = useSelectAll({
+ *       selectionModifiers,
+ *       sort: { field: 'title', order: 'ASC' },
+ *       filter: { title: 'foo' },
+ *   });
+ *   const handleClick = () => onSelectAll({
+ *       queryOptions: { meta: { foo: 'bar' } },
+ *       limit: 250,
+ *   });
+ *   return <Button onClick={handleClick}>Select All</Button>;
+ * };
+ */
+export const useSelectAll = ({
+    resource,
+    selectionModifiers,
+    sort,
+    filter,
+}: useSelectAllProps): ((options?: onSelectAllProps) => void) => {
+    const dataProvider = useDataProvider();
+    const queryClient = useQueryClient();
+    const notify = useNotify();
+
+    const onSelectAll = useCallback(
```

Comment:
I'd call it `handleSelectAll` instead. Same for the props. 

Also, it's a good use case for `useEvent` instead of `useCallback`.

Reply:
I don't agree for `useEvent` because it doesn't update the callback when dependencies change.
If I have more than 250 items and I change the `sort` or the `filter`, it will not select the 250 __first__ items.